packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -92,14 +96,21 @@ export class CDKDeployer implements BackendDeployer {
       // If we cannot load ts config, turn off type checking
       return;
     }
-    await this.executeChildProcess('npx', [
-      'tsc',
-      '--noEmit',
-      '--skipLibCheck',
-      // pointing the project arg to the amplify backend directory will use the tsconfig present in that directory
-      '--project',
-      dirname(this.backendLocator.locate()),
-    ]);
+    try {
+      await this.executeChildProcess('npx', [
+        'tsc',
+        '--noEmit',
+        '--skipLibCheck',
+        // pointing the project arg to the amplify backend directory will use the tsconfig present in that directory
+        '--project',
+        dirname(this.backendLocator.locate()),
+      ]);
+    } catch (err) {
+      throw new AmplifyUserError('SyntaxError', {
+        message:
+          'TypeScript validation check failed, check your backend definition',
+      });
```

Comment:
we should pass `err` as `cause` 
and we should do this everytime we're wrapping error.
bonus points if we can make linter rule for this.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

Reply:
Updated